tests/test_PessimisticPatternMatcherB3D.py

```diff
@@ -0,0 +1,273 @@
+
+    def testMatchSkip(self):
+        """ Test the ablity to skip specified patterns in the matching
+        process.
+        """
+        self.pyPPMb = PessimisticPatternMatcherB(
+            reference_catalog=self.reference_obj_array[:, :3],
+            log=self.log)
+
+        match_struct = self.pyPPMb.match(
+            source_catalog=self.source_obj_array, n_check=9, n_match=6,
+            n_agree=2, max_n_patterns=100, max_shift=60., max_rotation=5.0,
+            max_dist=5., min_matches=30, pattern_skip_array=np.array([0]))
+        self.assertEqual(len(match_struct.matches),
+                         len(self.reference_obj_array))
+        self.assertTrue(
+            np.all(match_struct.distances < 0.01/3600.0*__deg_to_rad__))
+
+    def testMatchMoreSources(self):
+        """ Test the case where we have more sources than references
+        but no rotation or shift.
+        """
+        self.pyPPMb = PessimisticPatternMatcherB(
+            reference_catalog=self.reference_obj_array[:500, :3],
+            log=self.log)
+
+        match_struct = self.pyPPMb.match(
+            source_catalog=self.source_obj_array, n_check=9, n_match=6,
+            n_agree=2, max_n_patterns=100, max_shift=60.0, max_rotation=5.0,
+            max_dist=5., min_matches=30, pattern_skip_array=None)
+        self.assertEqual(len(match_struct.matches),
+                         len(self.reference_obj_array[:500]))
+        self.assertTrue(
+            np.all(match_struct.distances < 0.01/3600.0 * __deg_to_rad__))
+
+    def testMatchMoreReferences(self):
+        """ Test the case where we have more references than sources
+        but no rotation or shift.
+        """
+        self.pyPPMb = PessimisticPatternMatcherB(
+            reference_catalog=self.reference_obj_array[:, :3],
+            log=self.log)
+
+        match_struct = self.pyPPMb.match(
+            source_catalog=self.source_obj_array[:500], n_check=9, n_match=6,
+            n_agree=2, max_n_patterns=100, max_shift=60., max_rotation=1.0,
+            max_dist=5., min_matches=30, pattern_skip_array=None)
+        self.assertEqual(len(match_struct.matches),
+                         len(self.reference_obj_array[:500]))
+        self.assertTrue(
+            np.all(match_struct.distances < 0.01/3600.0 * __deg_to_rad__))
+
+    def testShift(self):
+        """ Test the matcher when a shift is applied to the data. We say
+        shift here as while we are rotating the unit-sphere in 3D on our
+        'focal plane' this will appear as a shift.
+        """
+        self.pyPPMb = PessimisticPatternMatcherB(
+            reference_catalog=self.reference_obj_array[:, :3],
+            log=self.log)
+        theta = np.radians(45.0 / 3600.)
+        cos_theta = np.cos(theta)
+        sin_theta = np.sin(theta)
+        theta_rotation = np.array(
+            [[cos_theta, -sin_theta, 0.],
+             [sin_theta,  cos_theta, 0.],
+             [       0.,         0., 1.]])
```

Comment:
You use this more than once. How about adding a method to compute it? Similarly for some other rotation matrices.